common/lib/plugins/strategy/fastest_response/fastest_response_strategy_plugin.ts

```diff
@@ -0,0 +1,131 @@
+  private randomHostSelector: RandomHostSelector = new RandomHostSelector();
+
+  constructor(pluginService: PluginService, properties: Map<string, any>, hostResponseTimeService?: HostResponseTimeService) {
+    super();
+    this.pluginService = pluginService;
+    this.properties = properties;
+    this.hostResponseTimeService =
+      hostResponseTimeService ??
+      new HostResponseTimeServiceImpl(pluginService, properties, WrapperProperties.RESPONSE_MEASUREMENT_INTERVAL_MILLIS.get(this.properties));
+    this.cacheExpirationNanos = BigInt(WrapperProperties.RESPONSE_MEASUREMENT_INTERVAL_MILLIS.get(this.properties) * 1_000_000);
+  }
+
+  public getSubscribedMethods(): Set<string> {
+    return FastestResponseStrategyPlugin.subscribedMethods;
+  }
+
+  async connect(
+    hostInfo: HostInfo,
+    props: Map<string, any>,
+    isInitialConnection: boolean,
+    connectFunc: () => Promise<ClientWrapper>
+  ): Promise<ClientWrapper> {
+    const result = await connectFunc();
+    if (isInitialConnection) {
+      this.hostResponseTimeService.setHosts(this.pluginService.getHosts());
+    }
+    return result;
+  }
+
+  async forceConnect(
+    hostInfo: HostInfo,
+    props: Map<string, any>,
+    isInitialConnection: boolean,
+    forceConnectFunc: () => Promise<ClientWrapper>
+  ): Promise<ClientWrapper> {
+    const result = await forceConnectFunc();
+    if (isInitialConnection) {
+      this.hostResponseTimeService.setHosts(this.pluginService.getHosts());
+    }
+    return result;
+  }
+
+  acceptsStrategy(role: HostRole, strategy: string) {
+    return FastestResponseStrategyPlugin.FASTEST_RESPONSE_STRATEGY_NAME === strategy;
+  }
+
+  getHostInfoByStrategy(role: HostRole, strategy: string, hosts?: HostInfo[]): HostInfo | undefined {
+    if (!this.acceptsStrategy(role, strategy)) {
+      logAndThrowError(Messages.get("FastestResponseStrategyPlugin.unsupportedHostSelectorStrategy", strategy));
+    }
+    // The cache holds a host with the fastest response time.
+    // If the cache doesn't have a host for a role, it's necessary to find the fastest host in the topology.
+    const fastestResponseHost: HostInfo = FastestResponseStrategyPlugin.cachedFastestResponseHostByRole.get(role);
+    if (fastestResponseHost) {
+      // Found the fastest host. Find the host in the latest topology.
+      const foundHost = this.pluginService.getHosts().find((host) => host === fastestResponseHost);
+      if (foundHost) {
+        // Found a host in the topology.
+        return foundHost;
+      }
+    }
+    // Cached result isn't available. Need to find the fastest response time host.
+    const calculatedFastestResponseHost: HostInfo[] = this.pluginService
+      .getHosts()
+      .filter((host) => role === host.role)
+      .map((host) => new ResponseTimeTuple(host, this.hostResponseTimeService.getResponseTime(host)))
+      .sort((a, b) => {
+        return a.responseTime - b.responseTime;
+      })
+      .map((host) => host.hostInfo);
```

Comment:
Since we only need 1 host I don't think we need this extra `map()` here and just call `calculatedHost.hostInfo` later.